src/server/hset_family.cc

```diff
@@ -26,6 +26,48 @@ using namespace facade;
 
 namespace {
 
+struct ScanOpts {
+  string_view pattern;
+  size_t limit = 10;
+
+  constexpr bool Matches(std::string_view val_name) const {
+    return pattern.empty() ? true
+                           : stringmatchlen(pattern.data(), pattern.size(), val_name.data(),
+                                            val_name.size(), 0);
+  }
+
+  static OpResult<ScanOpts> TryFrom(CmdArgList args);
+};
+
+OpResult<ScanOpts> ScanOpts::TryFrom(CmdArgList args) {
```

Comment:
I just like the name for this, I think it make it clear, if they have a good naming in Rust then we can use it, and maybe I was using it so many times I got use to it. I have to say there are same good think about it. And for someone who don't know Rust it would not mean much. You should not use constructor to "convert" in this case, as this can fail. So I had to come up with a function that do this "creation", This is where I come from.

Reply:
It's great, I was joking :)

Reply:
but you need to resolve the conflicts to proceed.

Reply:
note - I added in the last push (other then resolve the issues), more cases for the unit tests, I know you're joked about this, just this is one feature I really liked there (safe casting)